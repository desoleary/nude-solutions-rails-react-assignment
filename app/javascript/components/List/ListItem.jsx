import { List, Typography } from 'antd';
import { arrayOf, node, number, oneOfType, string } from 'prop-types';
import React from 'react';

const { Item } = List;
const { Text } = Typography;

Item.displayName = 'ItemAntd';

const ListItem = ({ text, extra, actions, id }) => {
  const key = `list-item-${id}`;

  return (
    <Item extra={extra} actions={actions} key={key}>
      <Text>{text}</Text>
    </Item>
  );
};

ListItem.defaultProps = {
  actions: []
};

ListItem.propTypes = {
  id: oneOfType([number, string]).isRequired,
  text: string,
  extra: node,
  actions: arrayOf(node)
};

export default ListItem;
