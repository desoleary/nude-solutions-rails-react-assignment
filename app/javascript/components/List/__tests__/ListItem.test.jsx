import { List } from 'antd';
import { mount } from 'enzyme';
import React from 'react';
import ListItem from '../ListItem';

const { Item } = List;

describe('with full rendering', () => {
  let wrapper;
  const props = {
    id: 'some-identifier',
    text: 'some-text',
    extra: <div>some-extra-content</div>,
    actions: [<div>some-action-to-perform</div>]
  };

  beforeEach(() => {
    wrapper = mount(<ListItem {...props} />);
  });

  it('compares snapshot', () => {
    expect(wrapper.length).to.eql(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('passes props correctly', () => {
    const textContent = wrapper.text();

    expect(wrapper.find(Item).key()).to.eql(`list-item-${props.id}`);
    expect(textContent).to.include(props.text);
    expect(textContent).to.include('some-extra-content');
    expect(textContent).to.include('some-action-to-perform');
  });
});
