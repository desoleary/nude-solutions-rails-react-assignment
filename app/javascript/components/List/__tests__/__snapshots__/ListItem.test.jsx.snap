// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with full rendering compares snapshot 1`] = `
<ListItem
  actions={
    Array [
      <div>
        some-action-to-perform
      </div>,
    ]
  }
  extra={
    <div>
      some-extra-content
    </div>
  }
  id="some-identifier"
  text="some-text"
>
  <ItemAntd
    actions={
      Array [
        <div>
          some-action-to-perform
        </div>,
      ]
    }
    extra={
      <div>
        some-extra-content
      </div>
    }
    key="list-item-some-identifier"
  >
    <li
      className="ant-list-item"
    >
      <Text>
        <Base
          component="span"
          ellipsis={false}
        >
          <LocaleReceiver
            componentName="Text"
          >
            <ResizeObserver
              disabled={true}
              onResize={[Function]}
            >
              <Typography
                className=""
                component="span"
                style={
                  Object {
                    "WebkitLineClamp": null,
                  }
                }
              >
                <span
                  className="ant-typography"
                  style={
                    Object {
                      "WebkitLineClamp": null,
                    }
                  }
                >
                  some-text
                </span>
              </Typography>
            </ResizeObserver>
          </LocaleReceiver>
        </Base>
      </Text>
      <ul
        className="ant-list-item-action"
        key="actions"
      >
        <li
          key="ant-list-item-action-0"
        >
          <div>
            some-action-to-perform
          </div>
        </li>
      </ul>
      <div
        key="extra"
      >
        some-extra-content
      </div>
    </li>
  </ItemAntd>
</ListItem>
`;
